autobuild/workflow.md

```diff
@@ -0,0 +1,195 @@
+```
+mark@x230:~/src/blockdx-configs$ cd autobuild
+mark@x230:~/src/blockdx-configs/autobuild$ python3 manifest-management.py
+```
+
+**Output:**
+* manifest-latest.json
+
+**Warning:**
+this script is still WIP and not ready for use.
+
+
+
+# Workflow
+
+## Update coin configs
+**Audience:** everyone
+
+The new process to update a coin, eg: on a version upgrade is as follows:
+
+1. fork the https://github.com/blocknetdx/blockchain-configuration-files repo
+2. clone the repo to your local machine
+3. checkout a branch from master with a suitable name, eg: bump-thing-v4.2.2 
+4. update information in the manifest-latest.json
+5. update information in autobuild/configs/thing.base.j2
+6. run app.py as described above to update the {wallet,xbridge}-confs/thing.conf
+7. commit the changes to
+   * manifest-latest.json
+   * autobuild/configs/thing.base.j2
+   * wallet-confs/thing.conf
+   * xbridge-confs/thing.conf
+8. push the branch back to your repo
+
+Other parts of the workflow depend on who is performing them and take place in other repos but are documented here for simplicity in the order they are expected to happen.
+
+## Build a new Docker image
+**Audience:** 
+Not applicable to 3rd party developers who are simply looking to list a coin.
+
+* In the https://github.com/blocknetdx/dockerimages repo go to Actions and choose the BUILD IMAGE FROM TEMPLATE workflow. 
+* Click the *Run workflow* dropdown.
+* Choose the appropriate branch (most likely master) in the *Use workflow from* dropdown.
+* Enter the URL of your branch in the *Full path of a branch where the config can be found.* field, eg: https://raw.githubusercontent.com/walkjivefly/blockchain-configuration-files/bump-BLOCK-v4.3.3
+* Enter the full name of the coin you wish to create a Docker image for in the *The full name of a wallet. Example - bitcoin, dash, blocknet.* field
+* Enter the version you wish to build in *Tag or version of an image, please check manifest-latest.json. By default it takes the newest version.*
+* Click the green *Run workflow* button.
+
+If successful, the workflow takes around 30 minutes to build and push a new image to Docker hub. The image will have a -staging tag at this point signifying it has not been tested and released for public use.
+
+## Build exrproxy-env deployment package
+**Audience:**
+Not applicable to 3rd party developers who are simply looking to list a coin.
+
+* These steps are based closely on https://docs.blocknet.co/service-nodes/setup/#prepare-to-deploy-service-node.
+* Clone the https://github.com/blocknetdx/exrproxy-env repo to your local machine. You need to create a deploy script for the servicenode which will oversee your test trades, and a deploy script for the two trading nodes.
+* cd exrproxy-env/autobuild
+* cp examples/alldaemons.yaml custom.yaml
+* Edit custom.yaml to create a (testnet) servicenode which includes the coin to be tested and any counter-party coin you want to use.
```

Comment:
It makes sense for new coins to be tested on a `testnet` service node. However, we can't use the autobuild tools to deploy a testnet SNode until this issue is fixed: https://github.com/blocknetdx/exrproxy-env/issues/46
Feel like debugging that one?

Reply:
I haven't seen that error. My testnet servicenode was setup legacy-style. My attempts at adapting the exrproxy-env app.py to create a testnet trading node have not yet been successful.  I'll look into the testnet servicenode setup as well.